### PR TITLE
feat(chore): add SCA scans

### PR DESCRIPTION
**Pull request by sc979**
## Description

Implement SCA scan

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

### CODE REVIEW

File: .github/workflows/veracode-analysis.yml [hunk 1]
```diff
@@ -108,8 +111,7 @@ jobs:
           file: "${{ inputs.module_directory }}/${{ inputs.module_name }}-${{ github.sha }}-${{ github.run_id }}-veracode-binary.zip"
           baseline_file: "${{ env.baseline_file }}"
           create_baseline_from: "${{ env.create_baseline_from }}"
-          fail_build: false #TODO: Do not fail the jobs until grace period ends
-          #fail_build: '${{ needs.build.outputs.fail_build }}'
+          fail_build: '${{ needs.build.outputs.fail_build }}'
```
Comment (sc979) [inline .github/workflows/veracode-analysis.yml:114]: the output is set to false until the grace period end in lines 54 & 59

File: .github/workflows/veracode-analysis.yml [hunk 2]
```diff
@@ -140,20 +142,79 @@ jobs:
...
+              echo "[DEBUG] - Failure of '$ npm install'. Deleting the 'package.json' to avoid analysis failure"
+              rm -f package.json
+            else
+              echo "[DEBUG] - 'package-lock.json' ready for analysis"
+            fi
+          fi
+
+      - name: SCA scan
+        env:
+          SRCCLR_API_TOKEN: ${{ secrets.veracode_srcclr_token }}
```
Comment (paul-oureib) [inline .github/workflows/veracode-analysis.yml:209]: that may be a detail but: why is this value called "SRCCLR_API_TOKEN" in this worfklow's environment whereas it's called "veracode_srcclr_token" is other environments?

File: .github/workflows/veracode-analysis.yml [hunk 3]
```diff
@@ -140,20 +142,79 @@ jobs:
...
+          SRCCLR_REGION: "ER"
+
+        uses: veracode/veracode-sca@v1.09
+        # This action is based on following command:
```
Comment (paul-oureib) [inline .github/workflows/veracode-analysis.yml:213]:
```suggestion
        # This action is based on the following command:
```

File: .github/workflows/veracode-analysis.yml [hunk 4]
```diff
@@ -140,20 +142,79 @@ jobs:
...
           scantimeout: 60
           includenewmodules: true
           scanallnonfataltoplevelmodules: true
           deleteincompletescan: 1
           scanpollinginterval: 60 # time between two checks in seconds / [30 to 120]
+
+  sca-scan:
+    needs: [build]
+    name: Run a SCA scan
+    # only stable and unstable maintenances branches will produce a report
```
Comment (paul-oureib) [inline .github/workflows/veracode-analysis.yml:167]:
```suggestion
    # only stable and unstable maintenance branches will produce a report
```

File: .github/workflows/veracode-analysis.yml [hunk 5]
```diff
@@ -140,20 +142,79 @@ jobs:
...
+          SRCCLR_API_TOKEN: ${{ secrets.veracode_srcclr_token }}
+          SRCCLR_REGION: "ER"
+
+        uses: veracode/veracode-sca@v1.09
+        # This action is based on following command:
+        # curl -sSL https://download.sourceclear.com/ci.sh | sh -s -- scan "./${{ inputs.module_directory }}" --debug
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          create-issues: false
+          fail-on-cvss: 8
```
Comment (paul-oureib) [inline .github/workflows/veracode-analysis.yml:218]: is there a particular reason for placing the cursor to 8?

**Review by paul-oureib — APPROVED** [review]

**Review by tuntoja — APPROVED** [review]

**Pull request merged**